### ISSUE TITLE
Jam Scheduler seems to be stuck after one completed transaction

### ISSUE THREAD

**Issue by andrasfuchs** [joinmarket-webui/jam#391]
I started the Jam Scheduler about 10 hours ago with the default settings of 8 transactions over 8 hours. 

As far as I can tell the first transaction was completed successfully, but I keep getting this screen:
![image](https://user-images.githubusercontent.com/910321/178265435-73810fe6-46ab-4d18-9ac7-be58b127418a.png)

It looks like I'm stuck with the second transaction, but I can't seem to find the transaction id or anything that I can investigate.

What is the recommended action in these cases: should I wait even more or stop and restart the scheduler?
Is there a way to know what exactly happened that caused this?

**Comment by theborakompanioni** [comment]
Currently (and I am saying this for some time now), Jam is very light on user feedback. The only way to know what is going on is by inspecting the logs in `/root/.joinmarket/logs/jmwalletd_stdout.log` and/or `/root/.joinmarket/logs/jmwalletd_stderr.log` (on Umbrel).

Hope to have better feedback in upcoming versions, but it will still take some more time : /

Please, for the sake of your privacy, remove or change any sensitive information from your logs, if you post here. e.g. addresses, txids, timestamps, etc.

**Comment by andrasfuchs** [comment]
I waited another day without stopping the process, but I'm still stuck at the second transaction.

I found the log files under `~/umbrel/app-data/jam/data/joinmarket/logs`.

I can see a message that repeats in `jmwalletd_stdout.log` with different identifiers:
```
Directory 3...d.onion:5222 is telling us that J5...LW has left, but we didn't know about them. Ignoring.
```

And there are some errors in `jmwalletd_stderr.log`, but they don't have a timestamp, so I'm not sure if they are related to my issue:
```
jmclient.jsonrpc.JsonRpcError: {'code': -28, 'message': 'Verifying blocks…'}
Traceback (most recent call last):
  File "/src/scripts/jmwalletd.py", line 47, in <module>
    jmwalletd_main()
  File "/src/scripts/jmwalletd.py", line 26, in jmwalletd_main
    load_program_config(config_path=options.datadir)
  File "/src/jmclient/jmclient/configure.py", line 763, in load_program_config
    global_singleton.bc_interface = get_blockchain_interface_instance(
  File "/src/jmclient/jmclient/configure.py", line 898, in get_blockchain_interface_instance
    bc_interface = BitcoinCoreInterface(rpc, network,
  File "/src/jmclient/jmclient/blockchaininterface.py", line 174, in __init__
    blockchainInfo = self._rpc("getblockchaininfo", [])
  File "/src/jmclient/jmclient/blockchaininterface.py", line 234, in _rpc
    res = self.jsonRpc.call(method, args)
  File "/src/jmclient/jmclient/jsonrpc.py", line 165, in call
    raise JsonRpcError(response["error"])
jmclient.jsonrpc.JsonRpcError: {'code': -28, 'message': 'Loading P2P addresses…'}
```

I'm going to stop the Scheduler, and try again.

**Comment by andrasfuchs** [comment]
After I stopped the Scheduler, I got this message that prevents me from starting it again:
![image](https://user-images.githubusercontent.com/910321/178479612-ab849286-b3e5-4104-bd68-95e9ab68827b.png)

I have my sats only in my Jar 1, all others are empty.
![image](https://user-images.githubusercontent.com/910321/178480218-74a577a1-98f0-4b8c-ba2d-4802c4a68c81.png)

I'll try it again in about an hour.

**Comment by dnlggr** [comment]
Looks like we're [only checking Jar #0 for UTXOs](https://github.com/joinmarket-webui/joinmarket-webui/blob/master/src/components/Jam.jsx#L61) which might well be a bug. The scheduler can start from any jar IIRC. Do you know more @theborakompanioni?

**Comment by theborakompanioni** [comment]
> Looks like we're [only checking Jar #0 for UTXOs](https://github.com/joinmarket-webui/joinmarket-webui/blob/master/src/components/Jam.jsx#L61) which might well be a bug. The scheduler can start from any jar IIRC. Do you know more @theborakompanioni?

As far as I am aware, in its current state, the scheduler can and will always start from Jar #0.
There is the possibility to [provide options when starting the scheduler](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/api/wallet-rpc.yaml#L632-L651), but this is not released yet.

This is something to address till `v0.1.0` is released. It will require a fair amount of testing. Let's also keep in mind that starting from another Jar (aka `mixdepthsource`), `mixdepthcount` must also be adapted, otherwise additional Jars will be created. This is a rather uneducated statement and must be verified during development.

The current workaround is to send it to Jar#0 again. That's what I meant with
> It is safe to stop the scheduler, but some inconveniences will occur, e.g. funds in wrong jar, etc.

**Comment by dnlggr** [comment]
> There is the possibility to [provide options when starting the scheduler](https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/master/docs/api/wallet-rpc.yaml#L632-L651), but this is not released yet.

Oh yeah you're right. So currently you need to start the scheduler from Jar #0. This will change once the server project releases https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1260. I'll update the wording to make clear that funds need to be in Jar #0 for now.

**Comment by theborakompanioni** [comment]
This should be fixed once and for all after https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1347 has been merged and integrated with Jam v0.1.0.

Please reopen on demand if it ever happens again.
